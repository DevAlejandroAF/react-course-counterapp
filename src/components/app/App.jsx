import PropTypes from 'prop-types';

const App = function({ title }) {
  return (
    <>
      <h1>{ title }</h1>
    </>
  );
}

App.propTypes = {
  title: PropTypes.string.isRequired,
}

App.defaultProps = {
  title: "There's no title",
}

export default App;
