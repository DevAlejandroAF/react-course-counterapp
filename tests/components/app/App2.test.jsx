import { render, screen } from "@testing-library/react";
import App from "../../../src/components/app/App";

describe('Pruebas en <App />', () => {
  const title = 'Hello World';
  const subTitle = "There's no subtitle"
  test('Debe de mostrar el título en un H1', () => {
    const { container } = render(<App title={title} />);
    expect(container).toMatchSnapshot();
  });

  test('Debe de mostrar el mensaje "Hello World"', () => {
    render(<App title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
    // screen.debug();
  })

  test('Debe de mostrar el título en un h1', () => {
    render(<App title={title} />);
    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
  })

  test('Debe de mostrar el subtitulo enviado por props', () => {
    render(<App title={title} subTitle={subTitle} />);
    expect(screen.getAllByText(subTitle).length).toBe(2);
  })

});

