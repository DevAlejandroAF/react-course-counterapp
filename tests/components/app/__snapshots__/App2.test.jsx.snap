// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <App /> Debe de mostrar el t√≠tulo en un H1 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hello World
     
  </h1>
  <p>
    There's no subtitle
  </p>
  <p>
    There's no subtitle
  </p>
  <p>
    No name
  </p>
</div>
`;
